### PR TITLE
Build TPU package in CI

### PR DESCRIPTION
**Pull request by will-cromar**
It's not strictly just the "TPU" package since it can still use the GPU plugin. Will rename it when I get rid of the GPU-only build.

Move TPU CI into the main build/test workflow and consume the main packages. This should reduce queue times for the TPU CI, since builds won't occupy a TPU and prevent other tasks from running.

Next steps:

- Migrate GPU and CPU tests over to new dependency chain
- Fix whatever the doc push task is doing?
- Publish base image for upstream
- Remove old workflow and all old circle CI scripts :confetti_ball:

### CODE REVIEW

**Comment by will-cromar** [comment]
Successful TPU run here: https://github.com/pytorch/xla/actions/runs/8653732236/job/23730116979?pr=6918

**Comment by will-cromar** [comment]
I'll make ansible respect the torch pin in a follow-up. The current TPU CI also does not use the pin, so it's not a regression.

File: .github/workflows/_build_torch_xla.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+      - name: Setup gcloud
+        shell: bash
+        run: |
+          echo "${GCLOUD_SERVICE_KEY}" > $GOOGLE_APPLICATION_CREDENTIALS
+      - name: Checkout PyTorch Repo
+        uses: actions/checkout@v4
+        with:
+          repository: pytorch/pytorch
+          path: pytorch
+          # TODO: correct pin
```
Comment (JackCaoG) [inline .github/workflows/_build_torch_xla.yml:39]: are we checking out torch pin? I think this is a very important part of the CI..

**Comment by will-cromar** [comment]
Never mind, we did fix it in the current TPU CI. Do you want me to fix it in this PR or a follow up?

**Review by JackCaoG — APPROVED** [review]

**Comment by JackCaoG** [comment]
why is GPU build failing after 4s lol

**Comment by will-cromar** [comment]
Infrastructure issue? `Temporarily overriding HOME='/home/ec2-user/actions-runner/_work/_temp/aecbae00-42d3-4e19-adba-8747bad65ab6' before making global git config changes`

I'll try kicking it again. I am trying to speed up the CI, but not by _that_ much

**Pull request merged**